### DISCUSSION TITLE
How to list all packages installed as optional dependencies only?

### DISCUSSION THREAD

**Discussion by enihcam** [Morganamilo/paru#969] [Q&A]
I know `paru -Qdt` can list all dependency packages that are `neither required nor optionally required by any currently installed package`, but how to list all packages installed only as optional dependency but not required? Thanks.

**Comment by eclairevoyant** [comment]
Don't think `paru` is the tool for this. I would suggest installing `expac` instead.

If you want to print out all packages that are optdeps of some package:

```
comm -12 <(expac -l"\n" "%o" | sort -u) <(expac -l"\n" "%n\n%S" | sort -u)
```

If you want to print out all packages that are optdeps of some package _and_ not a hard dep of any package:

```
comm -12 <(expac -l"\n" "%o" | sort -u) <(expac -l"\n" "%n\n%S" | sort -u) | comm -23 - <(expac -l"\n" "%D" | sort -u)
```

Adapted from this SO answer (which answers kind of the opposite question) by `@Sunday`: https://unix.stackexchange.com/a/218990/36254
